### PR TITLE
Add cordon nodes before downscaling

### PR DESCRIPTION
**Pull request by fradee**
Review, please for solving https://github.com/kubernetes/autoscaler/issues/2868

### CODE REVIEW

**Comment by hato221** [comment]
@losipiuk Hi, any chance to review it? 🙏

File: cluster-autoscaler/main.go [hunk 1]
```diff
@@ -172,6 +172,7 @@ var (
 	balancingIgnoreLabelsFlag = multiStringFlag("balancing-ignore-label", "Specifies a label to ignore in addition to the basic and cloud-provider set of labels when comparing if two node groups are similar")
 	awsUseStaticInstanceList  = flag.Bool("aws-use-static-instance-list", false, "Should CA fetch instance types in runtime or use a static list. AWS only")
 	enableProfiling           = flag.Bool("profiling", false, "Is debug/pprof endpoint enabled")
+	cordonNodeBeforeTerminate = flag.Bool("cordon-node-before-terminate", false, "Should CA cordon nodes before termiante during downscale process")
```
Comment (mdkent) [inline cluster-autoscaler/main.go:175]: termiante -> terminating

**Comment by hato221** [comment]
@fradee Hi, did you find another workaround on how to achieve cordoning? 🙏

**Comment by nemo83** [comment]
I've just found out we are suffering the same issue. Are there alternatives out there?

**Comment by carlosjgp** [comment]
Hey @fradee thank you for the PR, this will be brilliant for a CI cluster where new PODs are scheduled to perform short/medium tasks... it's exactly what I need

I'm curious why this PR didn't go through... If you are busy or this is not a problem for you anymore, would you mind if I clone this PR?

**Pull request closed without merging**